app/src/main/java/io/halkyon/resource/page/CredentialResource.java

```diff
@@ -155,7 +155,8 @@ public Credential findByName(@PathParam("name") String name) {
         return credential;
     }
 
-    private void doUpdateCredential(Credential credential, CredentialRequest request) {
+    private Credential initializeCredential(CredentialRequest request) {
```

Comment:
Why not move this logic also into the service?

Reply:
Yes. I agree, indeed I was thinking exactly the same in my way to the train... :-)